### PR TITLE
[Network] Reclassify 'network nic scale-set' under 'vmss nic'

### LINKED ISSUE #782: [Network] Inconsistent command naming for NICs between VM and scale sets.
Currently we have nic operations for VMs under `az vm nic ...` but operations for nics in a scale set are located in `network nic scale-set ...`, which is not intuitive. Normally our scale-set equivalent commands are located under the top level `vmss` subcommand. 

Recommend we move the `az network nic scale-set` commands to `az vmss nic` for consistency.

Additionally, the parameter aliases should be made consistent with the rest of VMSS.

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #782. This places the scale-set nic operations under the "vmss" subcommand so that it is grouped logically with existing VMSS commands. It also updates the aliases to be consistent with those already in VMSS.

### CODE REVIEW

**Comment by yugangw-msft** [comment]
:shipit:

**Comment by derekbekoe** [comment]
Before this is checked in, should the commands not be moved to the vm command package then?

**Comment by derekbekoe** [comment]
Do we want to have vmss commands in the network package?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 1]
```diff
@@ -243,9 +243,9 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
     register_cli_argument('network nic {}'.format(item), 'load_balancer_inbound_nat_rule_ids', options_list=('--lb-inbound-nat-rules',), nargs='+', validator=validate_inbound_nat_rule_id_list, help='Space separated list of names or IDs of load balancer inbound NAT rules to associate with the NIC. If names are used, --lb-name must be specified.', completer=get_lb_subresource_completion_list('inboundNatRules'))
 
 # NIC ScaleSet
-register_cli_argument('network nic scale-set', 'virtual_machine_scale_set_name', options_list=('--vm-scale-set',), completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachineScaleSets'), id_part='name')
-register_cli_argument('network nic scale-set', 'virtualmachine_index', options_list=('--vm-index',))
-register_cli_argument('network nic scale-set', 'network_interface_name', id_part='child_name')
+register_cli_argument('vmss nic', 'virtual_machine_scale_set_name', options_list=('--vmss-name',), help='Scale set name.', completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachineScaleSets'), id_part='name')
+register_cli_argument('vmss nic', 'virtualmachine_index', options_list=('--vm-index',))
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: you can choose to do it at other time, but to be consistent with the existing vmss command, ,this should be renamed to `--instance-id`, which is what compute team recommended

**Comment by yugangw-msft** [comment]
Agreed with @Derekbekoe, we should move to VM unless there are non-trivial difficulties

**Comment by tjprescott** [comment]
Unless our VM package already has dependency on the network SDK, I would argue we not move them so as to not create an unnecessary dependency. If it already has the dependency, then there should be no harm in moving them. 

---
In reply to: [244237268](https://github.com/Azure/azure-cli/pull/808#issuecomment-244237268) [](ancestors = 244237268)

**Comment by yugangw-msft** [comment]
VM does have for [get ip address](https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py#L178) of vm

**Comment by tjprescott** [comment]
@derekbekoe @yugangw-msft comments addressed

**Comment by yugangw-msft** [comment]
:shipit:

**Pull request merged**